readthedocs/oauth/signals.py

```diff
@@ -13,11 +13,36 @@
 from readthedocs.oauth.notifications import MESSAGE_PROJECTS_TO_MIGRATE_TO_GITHUB_APP
 from readthedocs.oauth.tasks import sync_remote_repositories
 from readthedocs.projects.models import Feature
+from django.utils import timezone
 
 
 log = structlog.get_logger(__name__)
 
 
+@receiver(user_logged_in, sender=User)
+def save_last_login_method_used(sender, request, user, *args, **kwargs):
```

Comment:
Not documented, but there is a chance that this signal receives a `sociallogin` object as kwarg...